### DISCUSSION TITLE
Through mqtt remote switching, the long press and jog modes of tasmota's relay relay realize two functions. or other better way

### DISCUSSION THREAD

**Discussion by wei1980okb** [arendst/Tasmota#20190] [Q&A]
Recently I am learning tasmota rule and need to implement a function. But I don't know how to implement it and need help, thank you!
1. Basic situation:
Flashed the switch relay of tasmota's esp8266 and connected it to the MQTT server. It works normally.
2. Module settings:
D3 GPIO0 Relay 1
3. Set the jog mode:
The first way clicks for 0.5 seconds
PulseTime1 5

Now we want to realize that this esp8266 device has a relay that can be manually switched to realize the jog and long press functions at different times.
My idea is: after receiving the specified mqtt information a, automatically execute the following command on the console through the rules [PulseTime1 0 (restore default settings)] to change the jog mode to the long press mode
After receiving the specified mqtt information b, automatically execute the following command on the console through the rules [PulseTime1 5 (first jog for 0.5 seconds)] to change the long press mode to jog mode.

**Comment by sfromis** [comment]
You are not explaining what you're trying to solve, but [any Tasmota command can be published via MQTT](https://tasmota.github.io/docs/MQTT/#commands-over-mqtt) without needing to involve rules.

**Comment by wei1980okb** [comment]
According to your reminder, the problem has been solved, thank you very much!
![11111](https://github.com/arendst/Tasmota/assets/129461231/cfb38cd6-bf00-46dc-b7d4-c851fda2924e)

**Comment by wei1980okb** [comment]
Thank you
